libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -52,6 +51,6 @@ getDState :: NewEpochState era -> DState era
 getDState = certDState . lsCertState . esLState . nesEs
 
 queryConstitutionHash ::
-  EraGovernance era => NewEpochState era -> Maybe (SafeHash (EraCrypto era) ByteString)
+  EraGovernance era => NewEpochState era -> Maybe (SafeHash (EraCrypto era) (Constitution era))
```

Comment:
So, this is wrong for two reasons:

* This will break query serialization, thus breaking query interface
* Function is named `queryConstitutionHash`, but it returns the `Constitution`. 

What we want to do is leave the function type as it was before, make it return the hash and add a new query that returns the actual `Constitution`

Reply:
The `Constitution` happens to be a phantom type here, very confusing to notice :slightly_smiling_face:. But we don't want to break the API :100:  Will a single `castSafeHash` save the day here? I have pushed this change. Let me know what you think :raised_hands: